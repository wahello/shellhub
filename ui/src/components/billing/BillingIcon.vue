<template>
  <fragment>
    <font-awesome-icon
      v-if="isDefaultIcon()"
      icon="credit-card"
      size="lg"
      data-test="default-icon"
    />

    <font-awesome-icon
      v-if="!isDefaultIcon()"
      :icon="[ 'fab', icon() ]"
      size="lg"
      data-test="type-icon"
    />
  </fragment>
</template>

<script>

export default {
  name: 'BillingIcon',

  props: {
    iconName: {
      type: String,
      required: true,
    },
  },

  data() {
    return {
      cardIcon: {
        amex: 'cc-amex',
        dinersClub: 'cc-diners-club',
        discover: 'cc-discover',
        jcb: 'cc-jcb',
        mastercard: 'cc-mastercard',
        visa: 'cc-visa',
      },
    };
  },

  methods: {
    isDefaultIcon() {
      return this.cardIcon[this.convertIconName()] === undefined;
    },

    icon() {
      return this.cardIcon[this.convertIconName()] || 'credit-card';
    },

    convertIconName() {
      if (this.iconName === 'diners-club') {
        return 'dinersClub';
      }

      return this.iconName;
    },
  },
};

</script>
