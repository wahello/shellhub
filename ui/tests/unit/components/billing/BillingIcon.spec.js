import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
import BillingIcon from '@/components/billing/BillingIcon';

describe('BillingIcon', () => {
  const localVue = createLocalVue();
  localVue.use(Vuex);

  localVue.component('FontAwesomeIcon', FontAwesomeIcon);

  let wrapper;

  const iconName = 'cc-amex';
  const defaultIcon = 'credit-card';

  const cardIcon = {
    amex: 'cc-amex',
    dinersClub: 'cc-diners-club',
    discover: 'cc-discover',
    jcb: 'cc-jcb',
    mastercard: 'cc-mastercard',
    visa: 'cc-visa',
  };

  beforeEach(() => {
    wrapper = shallowMount(BillingIcon, {
      localVue,
      stubs: ['fragment'],
      propsData: { iconName },
    });
  });

  ///////
  // Component Rendering
  //////

  it('Is a Vue instance', () => {
    expect(wrapper).toBeTruthy();
  });
  it('Renders the component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  ///////
  // Data and Props checking
  //////

  it('Receive data in props', () => {
    expect(wrapper.vm.iconName).toBe(iconName);
  });
  it('Compare data with default value', () => {
    expect(wrapper.vm.cardIcon).toEqual(cardIcon);
  });

  //////
  // HTML validation
  //////

  it('Renders the template with data', () => {
    //////
    // In this case, the default icon is tested.
    //////

    wrapper = shallowMount(BillingIcon, {
      localVue,
      stubs: ['fragment'],
      propsData: { iconName: defaultIcon },
    });

    expect(wrapper.find('[data-test="default-icon"]').exists()).toBe(true);
    expect(wrapper.find('[data-test="type-icon"]').exists()).toBe(false);
  });

  //////
  // In this case, the other icons are tested.
  //////

  Object.keys(cardIcon).forEach((iconKey) => {
    wrapper = shallowMount(BillingIcon, {
      localVue,
      stubs: ['fragment'],
      propsData: { iconName: iconKey },
    });

    expect(wrapper.find('[data-test="default-icon"]').exists()).toBe(false);
    expect(wrapper.find('[data-test="type-icon"]').exists()).toBe(true);
  });
});
