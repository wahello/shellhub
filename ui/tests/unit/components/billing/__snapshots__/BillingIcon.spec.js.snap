// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BillingIcon Renders the component 1`] = `
<fragment-stub>
  <font-awesome-icon-stub icon="credit-card" size="lg" data-test="default-icon"></font-awesome-icon-stub>
  <!---->
</fragment-stub>
`;
